content/work/timedesigner.md

```diff
@@ -81,15 +114,403 @@ press:
+We end up with 1 column for each minute of the school day.
+
+```css
+.row-grid {
+  grid-template-columns: repeat(var(--day), minmax(1px, 1fr));
+}
+```
+
+{{ embed.figure(
+  data=scenariogrid,
+  caption='After creating a new scenario,
+    the start and end times are used to make the overall grid.
+    This example also shows vertical grid lines drawn with
+    CSS borders using two different colors for the hour
+    (`major`) and half hour (`minor`) segments.'
+) }}
+
+{{ embed.figure(
+  data=emptygridoverlay,
+  caption='We can see a great overview of the grid by
+    using the grid inspector tool in Firefox.'
+) }}
+
+Each experience (`.exp`) on the grid is placed
+using a start time (`--start`)
+and duration (`--span`).
+We include the option to add a transition period (`--plus`)
+as well.
+
+```html
+<div style="--start: 410; --span: 55; --plus: 10;" class="exp">
+```
+
+The duration and transition period are added to create a `--total`
+span amount for the grid-column.
+
+```css
+.exp {
+  --total: calc(var(--span) + var(--plus));
+  grid-column: calc(var(--start) + 1) / span var(--total);
+}
+```
+
+{{ embed.figure(
+  data=experiencegrid,
+  caption='Using the Inspector, we see how variables are defined and
+    used in the markup and css to place experiences anywhere on the grid.'
+) }}
+
+For optimal flexibility,
+an experience may overlap in time with another experience.
+Thanks to CSS Grid,
+we are able to place these in a sensible way with minimal effort.
+Using `grid-auto-flow` with a value of `dense`,
+we can allow the row to place experiences where they
+fit, regardless of where they were added in the markup.
+Additionally, we can assign the row a minimum height
+and allow it to grow taller as needed using `minmax`.
+
+```css
+.row-grid {
+  grid-auto-flow: dense;
+  grid-auto-rows: minmax(1.75rem, auto);
+}
+```
+
+{{ embed.figure(
+  data=overlapexperience,
+  caption='CSS Grid allows us to backfill available
+    space of items regardless of document ordering.
+    This can lead to accessibility issues if not applied with care.'
+) }}
+
+### Drag-and-Drop Schedule Interface with CSS Grid
+
+This use of CSS grid combined beautifully
+with an intuitive drag-and-drop interface.
+An experience's `--start` variable is
+updated continuously while being dragged,
+allowing it to move responsively along the grid.
+
+{{ embed.figure(
+  data=[{
+    img: 'work/timedesigner/dragdrop.jpg',
+    alt: 'close up of an experience being dragged on the scheduling grid'
+  }],
+  caption='Each rectangle -- or "experience" --
+    can be dragged and dropped,
+    updating the CSS variables.'
+) }}
+
+### Print-to-PDF Reports
+
+Users needed the ability to view, print or download
+PDF reports of their scenarios
+using print-specific layouts.
+To do this, we created
+custom print styles,
+and used print-to-pdf
+in a [headless Chrome](https://github.com/fate0/pychrome) instance
+to render an HTML string to a PDF.
```

Comment:
@jgerigmeyer do you think it is worth mentioning that we included a few ways to see the data in the report? 
![scroll-report](https://user-images.githubusercontent.com/1581694/132907803-6b40a2ec-c3a9-40bf-897f-e04ef452140f.gif)

Reply:
@SondraE Sure, I think that's a good idea -- feel free to add it 😄

Reply:
@jgerigmeyer I updated the image and added some text about the additional report display layouts.